src/github.ts

```diff
@@ -0,0 +1,85 @@
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import { GetResponseDataTypeFromEndpointMethod } from '@octokit/types';
+import { GitHub } from '@actions/github/lib/utils';
+import { extractBumpComment } from './common';
+
+export class Repo {
+  readonly octokit: InstanceType<typeof GitHub>;
+  readonly owner: string;
+  readonly name: string;
+  readonly prNumber?: number;
+
+  constructor() {
+    // Fetch GitHub Action context
+    // from GITHUB_REPOSITORY & GITHUB_EVENT_PATH
+    const { owner, repo, number: prNumber } = github.context.issue;
+    // const { sha, ref } = github.context;
+
+    this.owner = owner;
+    this.name = repo;
+    this.prNumber = prNumber;
+
+    this.octokit = this.getOctokit();
+  }
+
+  isPr(): boolean {
+    return !!this.prNumber;
+  }
+
+  getOctokit(): InstanceType<typeof GitHub> {
+    const ghToken = core.getInput('github-token') || process.env['GITHUB_TOKEN'];
+
+    if (!ghToken) {
+      throw new Error(
+        'No GITHUB_TOKEN env variable available. Are you sure to run this package from a Github Action?',
+      );
+    }
+
+    return github.getOctokit(ghToken);
+  }
+
+  async createOrUpdateComment(body: string, digest: string): Promise<void> {
+    const { owner, name: repo, prNumber: issue_number, octokit } = this;
```

Comment:
Where does this `issue_number` come from?

Reply:
this instantiation syntax is to **rename keys** from a hash:

```javascript
const hash = { a: "a key", b: "b key" };
const { a, b: renamedB } = hash;

console.log(a);
=> "a key"
console.log(renamedB);
=> "b key"
```

In our case it basically creates a variable called `issue_number` containing the value of `this.prNumber`. And I named it `issue_number` because it's the expected parameter name from the Github functions: https://octokit.github.io/rest.js/v18#issues-list-comments.

Altogether this is sugar syntax to avoid having to repeat `issue_number: this.prNumber` in each github function call. But if you find it clearer to read I can rewrite the whole function with more “explicit” parameter passing.

Reply:
No no it's OK, I just need to spend more time learning TS! Thanks for the explanation :+1: